### ISSUE TITLE
"slick-current" is always on first slide despite initialSlide != 0

### ISSUE THREAD

**Issue by adroste** [akiran/react-slick#1874]
v0.27.9

If I set initialSlide to something != 0, the internal index as well as the css class "slick-center" are set correctly.
However, the css class "slick-current" is on the wrong element. It is always on the first one.
In the example picture below the slider was rendered with initialSlide = 2.

![Screenshot 2020-08-17 at 18 54 33](https://user-images.githubusercontent.com/18649298/90422397-18ee5480-e0bb-11ea-929b-5c9c098a71a5.png)

**Comment by akiran** [comment]
Thanks for reporting. I will fix it in a day or two

**Comment by NikolaStanisavljevic** [comment]
@akiran 
Are there any updates on this, i've encountered same problem on same version (v0.27.9)?

**Comment by alexdroste** [comment]
@NikolaStanisavljevic 
The workaround I use right now is to not make my styling dependent on slick-current. If I just ignore it and use a custom-class or even slick-center the styling works just fine.

**Comment by NikolaStanisavljevic** [comment]
@alexdroste
 Thank you. but my But my issue is not related to styling of slider. Setting `initialSlide` on page load causes sliding bug because "slick-current" is on the wrong element.

**Comment by morganHipay** [comment]
This bug appears only when you set `infinite` to false, with true value, it works. Hope it helps you

**Comment by NikolaStanisavljevic** [comment]
@morganHipay Yeah. That seems to be the problem. Thank you.

**Comment by KarenQiao** [comment]
any update? same issue here

**Comment by hungdev** [comment]
i faced same issue :/ anybody can help me?

**Comment by GaneshKosuri** [comment]
i faced same issue

**Comment by AceBerry924** [comment]
@akiran , is there any solution for this?
Cause I got this error too...
Thank you

**Comment by tanyaburlakova** [comment]
@akiran hello! any chance to fix it? This problem is still alive.

**Comment by hayk-front** [comment]
> Thanks for reporting. I will fix it in a day or two

Is it fixed on some version? I used 0.28.1 upgraded to latest 0.29, still get the problem.

**Comment by eigood** [comment]
```
    _this.state = _objectSpread(_objectSpread({}, _initialState["default"]), {}, {
      currentSlide: _this.props.initialSlide,
      targetSlide: _this.props.infinite ? undefined : _this.props.initialSlide,
      slideCount: _react["default"].Children.count(_this.props.children)
    });

```

I hot-fixed my node_modules, and added the targetSlide line.  I can't currently check this out and build the package, as I do not have node.js installed into my desktop(I run everything via containers, so that my desktop can stay clean).

I was experiencing this problem with infinite=false, then using paging, and setting initialSlide=8 when slidesToScroll=4.  Clicking previous would then jump to the very first page.

**Comment by ko-devHong** [comment]
> 

same issue

**Comment by asad-makhdoom** [comment]
I was experiencing same issue and I figured out that that it came due to version upgrade from `0.25.2` to `0.29.0`.

On version `0.25.2`, it puts both `slick-center` and `slick-current` classes on initial slide and moves both classes to the selected slide whenever the selection is changed. So if you style the selected slide using `slick-current` class, it was totally fine on older version.

But in newer version, `0.29.0` in my case, this implementation has been changed. Now `slick-current` is always pointing to first slide even if you change the initial slide, but `slick-center` class is always on correct place (initial slide or selected slide).

**Solution:**
Instead of `slick-current` class, use `slick-center` class to style initial or selected slide.
